hal/src/thumbv7em/icm.rs

```diff
@@ -0,0 +1,1640 @@
+    /// found in main or secondary list.
+    #[inline]
+    pub
+    get_rhien, set_rhien: 4;
+    /// Digest Mismatch Interrupt Disable
+    ///
+    /// * true: the `RBE` flag remains cleared even if the setting condition is
+    /// met.
+    /// * false: the `RBE` flag is set when the hash value just
+    /// calculated from the processed region differs from expected hash
+    /// value.
+    #[inline]
+    pub
+    get_dmien, set_dmien: 5;
+    /// Bus Error Interrupt Disable
+    ///
+    /// * true: the flag remains cleared even if the setting condition is met.
+    /// * false: the flag is set when an error is reported on the sysstem bus
+    /// by the bus MATRIX.
+    #[inline]
+    pub
+    get_beien, set_beien: 6;
+    /// Wrap Condition Interrupt Disable
+    ///
+    /// * true: the `RWC` flag remains cleared even if the setting condition is
+    /// met.
+    /// * false: the `RWC` flag is set when the WRAP is encountered.
+    #[inline]
+    pub get_wcien, set_wcien: 7;
+    /// End Bit Condition Interrupt Enable
+    ///
+    /// * true: the `REC` flag remains cleared even if the setting condition is
+    /// met.
+    /// * false: the `REC` flag is set when the descriptor having the
+    /// `EOM` bit set is processed.
+    #[inline]
+    pub get_ecien, set_ecien: 8;
+    /// Monitoring Status Updated Condition Interrupt Enable
+    ///
+    /// * true: the `RSU` flag remains cleared even if the condition is met.
+    /// * false: the `RSU` flag is set when the corresponding descriptor is
+    /// loaded from memory to ICM.
+    #[inline]
+    pub get_suien, set_suien: 9;
+    /// Processing Delay
+    ///
+    /// Allows setting short or long delay.
+    ///
+    /// See [`Procdly`]
+    #[inline]
+    pub get_procdly, set_procdly: 10;
+    get_algo_bits, set_algo_bits: 14, 12;
+}
+
+impl RegionConfiguration {
+    const fn default() -> Self {
+        //RegionConfiguration { rcfg: 0x3F0 }
+        RegionConfiguration(0x3F0)
+    }
+}
+
+impl RegionConfiguration {
+    /// User SHA Algorithm
+    ///
+    /// Allow setting this regions [`icm_algorithm`].
+    #[inline]
+    pub fn set_algo(&mut self, value: icm_algorithm) {
+        self.set_algo_bits(value.into());
+    }
+
+    /// User SHA Algorithm
+    ///
+    /// Get the current user sha algorithm
+    #[inline]
+    pub fn get_algo(&mut self) -> icm_algorithm {
+        match self.get_algo_bits() {
+            2 => icm_algorithm::SHA224,
+            4 => icm_algorithm::SHA256,
+            _ => icm_algorithm::SHA1,
+        }
+    }
+    /// Reset the [`RegionConfiguration`] to default values
+    ///
+    /// Useful if changing between hashing and monitoring, etc.
+    #[inline]
+    pub fn reset_region_configuration_to_default(&mut self) {
+        self.0 = 0x3F0_u32;
+    }
+}
+
+/// Processing Delay
+///
+/// For a given SHA algorithm, the runtime period has two possible lengths
+pub enum Procdly {
```

Comment:
This name feels a bit out of place now. If `ProcessingDelay` is too long, I would at least capitalize the `D` to `ProcDly`.

Reply:
Adopted :+1: `Procdly` -> `ProcessingDelay`